### ISSUE TITLE
bug(cli): analyze --bogus throws an unhandled exception instead of rejecting the unknown option

### ISSUE THREAD

**Issue by jchable** [jchable/gpx-utility-analyzer#136]
## Problem

`gpx-analyzer analyze --bogus` does not produce a clean error. It terminates with an **unhandled exception and a stack trace**.

## Cause

`analyze`'s `files` argument is declared with `ArgumentArity.OneOrMore`. An unrecognised token like `--bogus` is therefore not rejected by the parser as an unknown option — it is bound as an argument *value*. The token then reaches `FileResolver`, which throws:

```
--bogus is not a .gpx file
```

as an unhandled exception rather than a handled diagnostic.

Contrast with `benchmark --bogus`, whose argument has single arity: there the parser rejects the token properly, with `Unrecognized command or argument '--bogus'.` and exit code 1.

## Impact

- A simple typo (`--formt json`, `--presset trail`) produces a stack trace instead of a usable message — poor UX for a CLI whose primary audience is scripting.
- The distinction is invisible to users: the same mistake behaves completely differently depending on which sub-command they typed.
- `split` and `merge` should be checked too; `merge` also uses `OneOrMore`.

## Fix

Validate resolved tokens before they reach `FileResolver`, and treat an argument value beginning with `--` as an unrecognised option rather than a filename. Emit the same shape of error the parser produces for single-arity commands, and exit non-zero without a stack trace.

Whatever the mechanism, the CLI should never surface an unhandled exception for malformed input — every bad-input path should produce a one-line diagnostic on stderr and a non-zero exit code.

## Notes

This behaviour is pinned by a characterization test (`CliParseTests`) added alongside the System.CommandLine 2.x migration, so the current behaviour is locked in and any change will be visible. That test asserts the message verbatim and will need updating as part of this fix — the message it pins is this codebase's own, not System.CommandLine's.

## Acceptance criteria

- [ ] `analyze --bogus` exits non-zero with a one-line error naming the unrecognised token, no stack trace.
- [ ] Same for `merge` and `split`.
- [ ] A genuinely mistyped filename (`trakc.gpx`) still reports a *file not found* style error, distinct from an unrecognised option.
- [ ] The characterization test is updated to the new expected output.

**Comment by jchable** [comment]
Fixed and merged to `main` via #145 (or the follow-up commits on `dev`). The commits referenced this issue number without a closing keyword, which is why it stayed open.
